resource_retriever/CMakeLists.txt

```diff
@@ -12,8 +12,7 @@ endif()
 
 find_package(ament_cmake_ros REQUIRED)
 find_package(ament_index_cpp REQUIRED)
-# TODO(wjwwood): remove libcurl_vendor and just use system curl when possible
-find_package(libcurl_vendor REQUIRED)
+find_package(CURL REQUIRED)
```

Comment:
I'm fairly certain this is going to fail on Windows.  At least, it did for me locally.  I had to do the following:

1. `> set CMAKE_PREFIX_PATH=C:\ProgramData\chocolatey\lib\curl\tools\curl-7.67.0-win64-mingw`
2. Add `list(APPEND CMAKE_FIND_LIBRARY_SUFFIXES ".dll.a" ".a")`

With both of those things, I was able to compile successfully.  However, the tests failed because they couldn't then find the dll at runtime.  That *might* be able to be solved by putting the environment hooks for Windows back in place, but I'm not entirely sure.

Reply:
I should also note that I got that `list` hint from: https://discourse.cmake.org/t/problem-with-findpackage-curl-required-and-precompiled-curl-7-68-0-for-windows/686/2

Reply:
Hmm, yeah adding `C:\ProgramData\chocolatey\lib\curl\tools\curl-7.67.0-win64-mingw` to `PATH` works as well. It's not very portable though, since the path changes with the version of curl (e.g. I have 7.77.0). I couldn't find any recommended solutions involving CMake online... maybe this is the motivation to keep our vendor package.